src/NodaTime/Interval.cs

```diff
@@ -155,6 +155,14 @@ public Instant End
         [Pure]
         public bool Contains(Instant instant) => instant >= start && instant < end;
 
+        /// <summary>
+        /// Returns whether or not this interval intersects with the given interval.
+        /// </summary>
+        /// <param name="interval">Interval to test.</param>
+        /// <returns>True if this interval intersects with the given interval; false otherwise.</returns>
+        [Pure]
+        public bool Intersects(Interval interval) => interval.start < end && interval.end > start;
```

Comment:
Would it be useful to change this to an `Intersection` method that returns a `Nullable<Interval>` I wonder?

Reply:
If you're going to add that, it's probably worth adding union and perhaps difference/symmetric difference.